flux-pipe/plotting/plot_fieldmap.py

```diff
@@ -0,0 +1,280 @@
+                   vmin=vmin, vmax=vmax)
+
+        # Plot all the fluxons
+        if plot_all:
+            colors_all = ['orange' if s > 0 else 'teal' for s in f_sgn]
+            ax0.scatter(f_lon, f_lat, s=3**2, c=colors_all, alpha=0.6)
+
+        # Plot the open fluxons
+        if plot_open:
+            colors_open = ['red' if s > 0 else 'blue' for s in oflx_low]
+            # sc00 = ax0.scatter(np.deg2rad(olon_low), np.sin(olat_low), c=oflx_low, cmap="PiYG", alpha=0.75, label='B(1.0R$_\odot$)')
+            # sc00 = ax0.scatter(th_clow, ph_clow, c=cflx_low, cmap="PuOr_r", alpha=0.55, label='B(1.0R$_\odot$)')
+            sc00 = ax0.scatter(th_olow, ph_olow, s=5**2, c=colors_open, alpha=1.0, label='Open Fields', edgecolors='k')
+
+        if ax is None:
+            shp = magnet.shape #pixels
+            plt.axis('off')
+            sz0=6 #inches
+            ratio = shp[1]/shp[0]
+            sz1=sz0*ratio #inches
+            DPI = shp[1] / sz1 #pixels/inch
+            fig.set_size_inches((sz1, sz0))
+            plt.tight_layout()
+            # print(f"\n\t\tSaving {shorten_path(fluxon_map_output_path)}...")
+            # plt.savefig(fluxon_map_output_path, bbox_inches='tight', dpi=4*DPI)
+            # print(f"\t\tSaving {shorten_path(fluxon_map_output_path_top)}...")
+            plt.savefig(fluxon_map_output_path_top, bbox_inches='tight', dpi=4*DPI)
+            # plt.show()
+            plt.close(fig)
+    else:
+        if do_print: 
+            print(f"\tSkipped! Files already exist:")
+            print(f"\t\t{shorten_path(fluxon_map_output_path)}")
+            print(f"\t\t{shorten_path(fluxon_map_output_path_top)}")
+    if do_print: print(f"\n\t    n_open: {n_open}, n_closed: {n_closed}, n_total: {n_flux}, n_all: {fnum}, n_outliers: {n_outliers}")
+    
+    if do_print_top: 
+        print("\t\t    Success!")
+        print("\t\t\t```````````````````````````````\n\n")
+    return n_open, n_closed, n_flux, fnum, n_outliers
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='This script plots the expansion factor of the given radial_fr.dat')
+    parser.add_argument('--cr', type=int, default=None)
+    parser.add_argument('--nwant', type=int, default=None)
+    parser.add_argument('--open', type=str, default=None)
+    parser.add_argument('--closed', type=str, default=None)
+    parser.add_argument('--dat_dir', type=str, default='/Users/cgilbert/vscode/fluxon-data', help='data directory')
+    parser.add_argument('--batch', type=str, default="default_batch", help='select the batch name')
+    args = parser.parse_args()
+    batch = args.batch
+
+    (hdr, cr, fname, adapt, doplot, reduce) = load_magnetogram_params(args.dat_dir)
+    CR = args.cr or cr
+    n_open, n_closed, n_flux, fnum, n_outliers = magnet_plot(CR, args.dat_dir, batch, args.open, args.closed, do_print=True, reduce=reduce, nwant=args.nwant, do_print_top=True)
+    
+    # exit()
+
+# print("DONE!")
```

Comment:
Dead code below

Reply:
What is dead apparently can never die.

Reply:
this is somehow an old version of this file. The current version is significally improved